src/vs/workbench/services/textfile/common/textFileEditorModelManager.ts

```diff
@@ -217,6 +233,12 @@ export class TextFileEditorModelManager implements ITextFileEditorModelManager {
 						break;
 				}
 			});
+
+			// Install model content change listener
+			this.mapResourceToModelContentChangeListener[resource.toString()] = model.onDidContentChange(() => {
+				const newEvent = new TextFileModelChangeEvent(model, StateChange.CONTENT_CHANGE);
```

Comment:
@Tyriar I do not see a reason to create an object here each time a character is typed. the untitled event just sends around its resource URI and I would do the same thing here for files.

Cannot say it often enough, this model content change event WILL FIRE like CRAZY. And perf is a hot topic after hot exit :)